### PR TITLE
Add StoreProduct Android Tests 

### PR DESCRIPTION
**Pull request by vegaro**
We were missing tests for the StoreProduct mapper

### CODE REVIEW

File: android/src/test/java/com/revenuecat/purchases/hybridcommon/mappers/StoreProductMapperTests.kt [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        val expected = "CAD"
+        stubStoreProduct(priceCurrencyCode = expected).map().let {
+            assertThat(it["currencyCode"]).isEqualTo(expected)
+        }
+    }
+
+    @Test
+    fun `maps discounts correctly`() {
+        stubStoreProduct().map().let {
+            assertThat(it["discounts"]).isEqualTo(null)
```
Comment (tonidero) [inline android/src/test/java/com/revenuecat/purchases/hybridcommon/mappers/StoreProductMapperTests.kt:62]: Hmm I don't think we have any tests where this is not null?

File: android/src/test/java/com/revenuecat/purchases/hybridcommon/mappers/StoreProductMapperTests.kt [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    @Test
+    fun `maps currency code correctly`() {
+        val expected = "CAD"
+        stubStoreProduct(priceCurrencyCode = expected).map().let {
+            assertThat(it["currencyCode"]).isEqualTo(expected)
+        }
+    }
+
+    @Test
+    fun `maps discounts correctly`() {
```
Comment (tonidero) [inline android/src/test/java/com/revenuecat/purchases/hybridcommon/mappers/StoreProductMapperTests.kt:60]: Maybe rename to `maps null discounts correctly`?

File: android/src/test/java/com/revenuecat/purchases/hybridcommon/mappers/StoreProductMapperTests.kt [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    description: String = "A product description",
+    subscriptionPeriod: String? = "P1M",
+    freeTrialPeriod: String? = "P7D",
+    introductoryPrice: String? = null,
+    introductoryPriceAmountMicros: Long = 0,
+    introductoryPricePeriod: String? = null,
+    introductoryPriceCycles: Int = 0,
+    iconUrl: String = "http://url.com",
+    originalJson: JSONObject = JSONObject("{}")
+) = StoreProduct(
```
Comment (tonidero) [inline android/src/test/java/com/revenuecat/purchases/hybridcommon/mappers/StoreProductMapperTests.kt:111]: I guess once we merge the BC5 changes where this is an interface we will need to change this, but this works for now 👍

**Review by tonidero — APPROVED** [review]
LGTM!

File: android/src/test/java/com/revenuecat/purchases/hybridcommon/mappers/StoreProductMapperTests.kt (hunk 1 above)
Comment (vegaro) [reply android/src/test/java/com/revenuecat/purchases/hybridcommon/mappers/StoreProductMapperTests.kt:62]: oh true, good catch

Comment (vegaro) [reply android/src/test/java/com/revenuecat/purchases/hybridcommon/mappers/StoreProductMapperTests.kt:62]: oh, actually, it's always null in Android lol

**Pull request merged**